Automattic/wp-calypso
# Domains: Hide "Get authorization code" button if auth code is not required for TLD

## Proposed Changes

This PR
- Adds a `authCodeRequired` property to the domain objects and
- Hides the "Get authorization code" button in the domain transfer page if that property is false

### Screenshots

- Auth code required

>  ![Markup on 2023-11-29 at 15:54:05](https://github.com/Automattic/wp-calypso/assets/5324818/1dd463e5-e647-4e9e-ac0a-d8b08ef4421c)

- Auth code is not required

> ![Markup on 2023-11-29 at 15:55:12](https://github.com/Automattic/wp-calypso/assets/5324818/be472be1-ffc7-4126-8d92-eec5457a25fa)

## Testing Instructions

- This PR depends on D130380-code
- Open the live Calypso link or build this branch locally
- Go to a domain's transfer page and ensure the "Get authorization code" button is present
- Hack your backend to return `auth_code_required = false` for your domain's TLD
- Reload the domain transfer page and ensure the "Get authorization code" button is not shown

It's easier to hack the backend because the only current TLDs that do not require auth codes are `.uk`, `.io` and `.us`, and `.uk` has a different interface for domain transfers.

## Pre-merge Checklist
